File: javascript/src/services/costing.js
// Cost derivation (FSD §3.5) lives in the service layer, not the model or the store — the value is
// computed from the line items on demand, never stored (a 2.C decision we hold to). PartsCost for a
// work order = Σ(WorkOrderPart.Quantity × Part.UnitCost). TotalCost is finished in the DTO mapper.
import store from '../data/store.js';

export function partsCostFor(workOrderId) {
  const partsById = new Map(store.parts.map((p) => [p.id, p]));
  return store.workOrderParts
    .filter((wp) => wp.workOrderId === workOrderId)
    .reduce((sum, wp) => sum + wp.quantity * (partsById.get(wp.partId)?.unitCost ?? 0), 0);
}

File: dotnet/src/FleetLink.Api/Services/Costing.cs
using FleetLink.Api.Data;

namespace FleetLink.Api.Services;

/// <summary>
/// Cost derivation (FSD §3.5) lives in the service layer, computed from the line items on demand and
/// never stored. PartsCost for a work order = Σ(WorkOrderPart.Quantity × Part.UnitCost). TotalCost is
/// finished in the DTO mapper (LabourCost + PartsCost, or the frozen completion total for rule 11).
/// </summary>
public static class Costing
{
    public static decimal PartsCostFor(FleetStore store, Guid workOrderId) =>
        store.WorkOrderParts
            .Where(wp => wp.WorkOrderId == workOrderId)
            .Sum(wp => wp.Quantity * (store.Parts.FirstOrDefault(p => p.Id == wp.PartId)?.UnitCost ?? 0m));
}
